app/views/categories/index.html.erb

```diff
@@ -0,0 +1,21 @@
+<%= render partial: 'shared/navbar', locals: { title: 'Categories'} %>
+
+<div id="categories" class="">
+  <% if @categories.empty?%>
+    <h3 class='no-cat text-center fw-bold'>No Category Available</h3>
+  <% else %>
+    <% @categories.each do |category| %>
+    <br>
+    <div class='container d-flex flex-row justify-content-between p-3 align-items-center rounded shadow border-dark'>
+      <img src="<%= category.icon %>" alt=<%= category.name %>, class= 'img-fluid img-thumbnail w-25' />
+      <h2><%= link_to category.name, category_path(category.id), class: 'text-decoration-none fs-3' %></h2>
+      <p class='fw-semibold'>$<%= category.expenses.sum(:amount) %></p>
```

Comment:
:orange_book: [OPTIONAL] Your approach is working great, but there is a better approach that you can use. Your current approach will cost heavily in terms of resources as the database grows and the expenses list increases for a specific category. A better approach would be to create a total_amount column in the database that will store the total amount of expenses for a specific category. Then, you can add two methods to the Expense model that will be responsible for increasing and decreasing the total_amount value depending on whether an expense has been added or removed.

This approach will be more efficient as the database grows because it will only need to update the total_amount column when an expense is added or removed. This will prevent the database from having to recalculate the total_amount value for every expense in the category.

Here is an example of how you can implement this approach:

```
#Add a column to the category table
class AddTotalAmountToCategories < .......# inherit from 
  def change
    add_column :categories, :total_amount, :integer, default: 0
  end
end

class Expense < ApplicationRecord
  # ... Existing Code

  after_save :update_category_total_amount
  before_destroy :reduce_category_total_amount

  private

  def update_category_total_amount
    categories.each do |category|
      category.increment!(:total_amount, amount)
    end
  end

  def reduce_category_total_amount
    categories.each do |category|
      category.decrement!(:total_amount, amount)
    end
  end
end
```